src/catchup/CatchupManagerImpl.cpp

```diff
@@ -57,20 +56,31 @@ CatchupManagerImpl::processLedger(LedgerCloseData const& ledgerData)
 
     uint32_t lastReceivedLedgerSeq = ledgerData.getLedgerSeq();
 
-    // If LCL is already at-or-ahead of the ledger we just received from the
-    // network, we're up to date. Return early, nothing to do.
-    if (lastReceivedLedgerSeq <=
-        mApp.getLedgerManager().getLastClosedLedgerNum())
+    // 1. CatchupWork is not running yet
+    // 2. We've seen out of sync ledgers
+    // 3. The ledger just received is equal to lcl
+    // then it's possible we're back in sync and we can attempt to apply
+    // mSyncingLedgers
+    if (!mCatchupWork && !mSyncingLedgers.empty() &&
```

Comment:
If the condition is moved down, the `lastReceivedLedgerSeq <= mApp.getLedgerManager().getLastClosedLedgerNum()` condition (line 71) will also need to be changed to allow ledgers equivalent to lcl to flow through. I think this is an issue because we could potentially buffer a ledger equal to lcl while catchup is running. The `ledgerData.getLedgerSeq() <= config.toLedger()` condition (line 96) wouldn't help in a scenario where we are in the middle of applying buffered ledgers, which would take lcl past `config.toLedger()`. If a ledger equal to lcl came in at this point, we would buffer it.

This scenario sounds impossible (you'd have to externalize the same ledger twice), but I'm not sure if that assumption can be made. We could just duplicate the lcl check if catchup is running, but at that point I think what we have here is cleaner.